### PR TITLE
changes for admin dashbaord extention

### PR DESCRIPTION
**Pull request by JWittmeyer**
includes new field, new table & new view for labeling

PRs: 
frontend: https://github.com/code-kern-ai/refinery-ui/pull/55
submodule-model: https://github.com/code-kern-ai/refinery-submodule-model/pull/8
gateway: https://github.com/code-kern-ai/refinery-gateway/pull/43
updater: https://github.com/code-kern-ai/refinery-updater/pull/11
admin-dashboard: https://github.com/code-kern-ai/admin-dashboard/pull/45

### CODE REVIEW

File: src/app/base/components/comment/comment-helper.ts [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+    public static unregisterAllCommentRequests(caller: Object) {
+        if (CommentDataManager.commentRequests.has(caller)) {
+            CommentDataManager.commentRequests.delete(caller);
+        }
+    }
+
+    public static unregisterPartialCommentRequests(caller: Object, requests: CommentRequest[]) {
+        if (CommentDataManager.commentRequests.has(caller)) {
+            let comments = CommentDataManager.commentRequests.get(caller);
+            comments = comments.filter(c => requests.some(a => a.commentType == c.commentType && a.projectId == c.projectId && a.commentKey == c.commentKey));
```
Comment (jhoetter) [inline src/app/base/components/comment/comment-helper.ts:59]: personal opinion: don't like `a` and `c` as variable names (ok, `c` is fine, but `a` i don't get :D)

File: src/app/base/components/comment/comment.component.ts [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+  cData = {
+    comment: "test",
+    created_at: "2022-09-21T15:34:27",
+    created_by: "e1d5b42c-4ddf-472c-84bb-9685d784edbb",
+    creationUser: "Jens",
+    id: "8e7fa62e-bc17-41d7-b62c-1923517db588",
+    is_markdown: false,
+    is_private: false,
+    order_key: 1,
+    project_id: "8e7fa62e-bc17-41d7-b62c-1923517db589",
```
Comment (jhoetter) [inline src/app/base/components/comment/comment.component.ts:23]: @JWittmeyer i know that this is absolutely hardcoded as you were implementing it; this is just a note for you as a reminder to not forget about it :)

**Comment by jhoetter** [comment]
image
not 100% sure what that labeling task type is for. Is it meant to be displayed there?
- [ ] yes
- [x] no, i fixed it

_______________________________
damn good find :D when creating logic that you don't have to hardcode dropdown values bites you in the a... 
--> fixing it :)

**Review by jhoetter — APPROVED** [review]
phew, that is a big one :D but looks good! also, black-box testing was super w/o any errors. Really nice, especially the lock for crowd labelers, that worked like a charm with 2 windows opened :)

File: src/app/base/components/comment/comment-helper.ts (hunk 1 above)
Comment (JWittmeyer) [reply src/app/base/components/comment/comment-helper.ts:59]: would c and c2 or cc & ec be better? c is the comment from comments that are currently being filtered and c2 (a) is the one from the external/input array -> I'm not the biggest fan of long variable names for iteration variables (like list comprehension in python)

File: src/app/base/components/comment/comment.component.ts (hunk 2 above)
Comment (JWittmeyer) [reply src/app/base/components/comment/comment.component.ts:23]: wont :)

**Pull request merged**